File: front/app/types/physical/port.ts
import * as v from 'valibot'
import { connectionSchema } from '@/types/physical/connection'

// Port
export const portSchema = v.object({
    id: v.string(),
    name: v.pipe(v.string(), v.nonEmpty()),
    description: v.nullish(v.string()),
    serialNumber: v.nullish(v.string()),
    portTypeId: v.nullish(v.string()),
    connection: v.nullish(connectionSchema),
})

export const portFormSchema = v.partial(portSchema, ['id'])

// Export des types
export type PortType = v.InferInput<typeof portSchema>
export type PortTypeForm = v.InferInput<typeof portFormSchema>

File: api/app/graphql/app/physical/types/port.rb
# frozen_string_literal: true

module App
  module Physical
    module Types
      # PortType
      class Port < ::Types::BaseObject
        graphql_name 'PortTypeT'

        field :id, ID, null: true
        field :name, String, null: false
        field :port_type_id, ID, null: true
        field :description, String, null: true
        field :mac_address, String, null: true
        field :connection, Physical::Types::Connection, null: true
      end
    end
  end
end
